make kodi versioning simpler

Rechi opened the pull request:
With this change we don't have to update the return value in av_version_info after rebasing onto a new version.

Rechi commented:
@fritsch please have a look

fritsch commented:
What happens if someone builds his own ffmpeg and adds the "kodi" at the end?  He would end up without our fixes, right?

Rechi commented:
Normally `FFMPEG_VERSION` is defined to the version in the RELEASE file if root directory does not contain a .git directory.
If configured with `--extra-version` the submitted value gets appended to the `FFMPEG_VERSION`.
We are currently not configuring with this option, therefore `$(EXTRA_VERSION)` is empty and we pass Kodi as extra version.
Hopefully I've answered your question.

The reason why I've created this PR is, @FernetMenta rebased onto FFmpeg 3.3.1 but forgot to change the version at 73516fefff5c55a24e26f836ddd4ccc3a60a3ea0.
Or was this intention to not add the minor version?

FernetMenta commented:
I did not forget to update minor version. The mein reason why this version was hardcoded was that we are able to indentify our fork. Minor version is not that important in this case.

Rechi commented:
with this PR the version will be `#define FFMPEG_VERSION "3.3.1-Kodi"`
it's up to you if you want to keep it like it is currently or use this which doesn't need to update the version.

FernetMenta commented:
Of course I am fine with this change, it's good. It is much better than what we had. Thanks!

wsnipex committed:
only check for a git rev if the src tree is in a git repo

fixes the version string when building from the kodi depends src tree

Rechi committed:
Revert "Versioning: As we ship custom patches again directly tell with av_version_info"

This reverts commit 73516fefff5c55a24e26f836ddd4ccc3a60a3ea0.

Rechi committed:
common.mak: never ignore an error if strip doesn't succeed

Rechi committed:
use Kodi as extra version

Rechi commented:
rebased it on the release/3.4-kodi branch and removed the ability to specify `EXTRA_VERSION` as configure argument
If no objections, I'll merge this so that it is used after the next version bump.

MartijnKaijser commented:
@Rechi you should perhaps try to get parts of it upstreamed?
